bot/core/ratelimit.py

```
@@ -0,0 +1,114 @@
+        if not hasattr(ctx.bot, 'redis'):
+            logging.warning("Redis not initialized in bot, skipping rate limit check.")
+            return True
+
+        cooldown = RedisCooldown(rate, per, type)
+        key = cooldown.get_key(ctx)
+        redis_client: redis.Redis = ctx.bot.redis
+
+        # Use a Lua script for atomicity
+        # Keys: [key]
+        # Args: [rate, per]
+        lua_script = """
```

Ruff (0.14.6)

Local variable `lua_script` is assigned to but never used

Remove assignment to unused variable `lua_script`

(F841)